### PR TITLE
[ENH, MRG] Change n_points to signal_len

### PR DESCRIPTION
**Pull request by alexrockhill**
n_points isn't as intuitive and I would suggest representing this value in seconds.

### CODE REVIEW

File: examples/plot_phase_delay.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def clean_peak_finder(sig):
 def simulate_and_plot(phi_0, delay, ax, rng):
     """Simulate oscillations with frequency modulation"""
     # create the slow oscillations
-    time = np.arange(n_points) / fs
+    time = signal_len
```
Comment (TomDLT) [inline examples/plot_phase_delay.py]: This is not equivalent, `time` was an array and is now an int.

File: examples/plot_dar_model.py [hunk 2]
```diff
@@ -21,12 +21,12 @@
 low_fq = 5.0  # Hz
 low_fq_width = 1.0  # Hz
 
-n_points = 10000
+signal_len = 6.25  # sec
```
Comment (TomDLT) [inline examples/plot_dar_model.py]: 50 sec ?

**Comment by alexrockhill** [comment]
Ok, I'll make some changes. I this would make it much more intuitive to simulate evoked PAC because it makes much more sense to say the evoked potential onsets at 2 seconds and lasts 1 second or something like that and than to do it in points. I'll make it backward compatible though.

**Comment by TomDLT** [comment]
Thanks for the update. I think the key is to preserve backward compatibility, meaning that no user will see a change in their results. In your current proposition, the results are changed so I think we should try something else.

I would propose to keep the `n_points` positional parameter where it is, but to add a keyword parameter `signal_length=None`. Then, we can raise an error if both parameters are not None, or if both are None.

**Comment by alexrockhill** [comment]
Hmmm that is very clunky for the signal_len use case which would look like

```
simulate_pac(None, 200, 10, 50, 2.5, 0.4, signal_len=4)
```

I would recommend another way but I will push a commit that does that if you want to do that.

**Comment by TomDLT** [comment]
This is not ideal indeed, but it is transparent if you use kwargs, e.g.
```py
simulate_pac(signal_len=signal_len, fs=fs, high_fq=high_fq, low_fq=low_fq,
             low_fq_width=low_fq_width, noise_level=noise_level,
             n_points=n_points)
```
What do you think ?

**Comment by alexrockhill** [comment]
Your call if you want to merge it. I just think it's confusing and not very pythonic to have to pass `n_points=None` if you are using `signal_len`. If there was a way to be able to pass one or the other with kwargs I think that would be totally acceptable but with the required arguments that makes that not work. I think it would be better to use the non-backward compatible version with depreciation warnings or to just leave it as n_points.

**Comment by TomDLT** [comment]
Right, I will just leave it as `n_points` then.
Thanks for the suggestion though !

**Pull request closed without merging**